### ISSUE TITLE
Accessibility: Grouped form fields missing fieldset [Product Archive Rating Filter]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52128]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

These form fields do not always have a visible label that provides context to the group of inputs by default - the component inserts a heading block above it, but this can be deleted by the content creator. Additionally, there is not a programmatically associated label for the controls in the group, so screen reader users do not get the context for what the filters are for.

![Image](https://github.com/user-attachments/assets/302c2f6a-24a7-4182-99e4-9e9781af7c22)

### Component or Page: 

Component: Product Archive Rating Filter Block

### Name affected element:

Filter by Rating Block

### Relevant code snippets:

`<label for="5"><input id="5" class="wc-block-components-checkbox__input" type="checkbox" aria-invalid="false"><svg class="wc-block-components-checkbox__mark" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 20"><path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"></path></svg><span class="wc-block-components-checkbox__label"><div class="wc-block-components-product-rating"><div class="wc-block-components-product-rating__stars" role="img" aria-label="Rated 5 out of 5"><span style="width: 100%;">Rated <strong class="rating">5</strong> out of 5</span></div></div></span></label>`

### Recommended fix:

At a minimum, the checkboxes need to be wrapped in a <fieldset> tag and have a <legend> tag that describes the filters, such as <legend>Filter by $attribute_name</legend>. This can be screen reader only, but it would be best if it was visible on the page.  

A further enhancement on the block would be to include a heading tag in the legend and add a setting to the block that allows users to set the preferred heading level, H2-H6. Example: <legend><h2>Filter by $attribute_name</h2></legend> You want to allow users to be able to control the heading level as it may not always be correct for it to be set to an H2, however an H2 with be a good default. This can replace the inserted heading block whcih content creators can incorrectly remove. Or, if keeping the heading block either block the ability to remove it or warn users before they remove it. 

See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

**Comment by vikrampm1** [comment]
Closing this issue as per the discussion [here](https://github.com/woocommerce/woocommerce/pull/53681#issuecomment-2610425082).

**Comment by amberhinds** [comment]
@gigitux @vikrampm1 for the VPAT, do you know what block users should switch to instead of using this block?

**Comment by gigitux** [comment]
> [@gigitux](https://github.com/gigitux) [@vikrampm1](https://github.com/vikrampm1) for the VPAT, do you know what block users should switch to instead of using this block?

`Product Filters` block that will be available in WooCommerce 9.9 (next release).
